### PR TITLE
Cut over to use a pre-built image of 1.4.0-snap17

### PR DESCRIPTION
**Pull request by solarkennedy**
This makes 3 changes:
- Use prebuilt images instead of building our own (mostly because I couldn't get recent versions of marathon to build myself, so I gave up and just used their images. Plus it is faster)
- Use the latest version available (1.4.0-snap17)
- Make a wrapper so we get docker-compose logs if the tests fail for debugging.

ALSO THE TESTS PASS

Relates to https://github.com/mesosphere/marathon/issues/3783 of course. (for github cross-link)

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -24,7 +24,7 @@ mesosslave:
     - /var/run/docker.sock:/var/run/docker.sock
 
 marathon:
-  image: mesosphere/marathon:v1.4.0-snap17
+  image: mesosphere/marathon:v1.3.3
```
Comment (EvanKrall) [inline docker-compose.yml]: did you mean to set this to `mesosphere/marathon:MARATHON_VERSION`?

Comment (solarkennedy) [reply docker-compose.yml]: Whoops, I committed after the sed

**Comment by solarkennedy** [comment]
@EvanKrall eyeballing the logs on the jobs that are green, there are lots of lines with "New leader" in them from marathon, which makes me thing that the leader failover really is happening.

**Comment by solarkennedy** [comment]
Updated the build to include the most recent snapshot and marathon 1.3.6 cc @mattmb

**Pull request merged**